> I'm not confident that this will compile when merging to `master`. Correct me if I'm wrong, but the build process uses that branch for defining the CI/CD configuration, regardless of what's on the PR.
> 
> Unless you tested compiling the project off line using those versions of the JDK, I'll have to do that before we can be sure it won't break the build.

It is true that `ci.jenkins.io` will always use the Jenkinsfile from `main` /  `master`.
~~I did not check if the plugin will actually compile using JDK 17 / 21 however I can see that the `pom.xml` is pretty much up-to-date and thus there should not be any problem.~~ The builds will still generate Java 11 bytecode unless you decide to switch to the `2.479` baseline and `5.x` plugin-pom which will generate Java 17 bytecode.

Edit: I just checked, `mvn clean verify` fails on JDK 17 and 21 due to
```
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field protected transient int java.util.AbstractList.modCount accessible: module java.base does not "opens java.util" to unnamed module @38a8f1a9
```
amongst other similar errors. The culprit seems to be https://github.com/jenkinsci/adobe-cloud-manager-plugin/blob/master/pom.xml#L217-L226 which will override the `argLine` from the parent `plugin-pom`. Adding the parents property by prepending `${argLine}` to the configuration makes the build run just fine from what I can tell. See https://github.com/jenkinsci/adobe-cloud-manager-plugin/pull/111/commits/0d89309f819608927ea3d0dde1adbe48f187720c

--- a/pom.xml
+++ b/pom.xml
@@ -219,7 +219,7 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
                     <argLine>
-                        -javaagent:"${settings.localRepository}"/org/jmockit/jmockit/${jmockit.version}/jmockit-${jmockit.version}.jar
+                        ${argLine} -javaagent:"${settings.localRepository}"/org/jmockit/jmockit/${jmockit.version}/jmockit-${jmockit.version}.jar
                     </argLine>
                     <useSystemClassLoader>false</useSystemClassLoader>
                 </configuration>

